apps/src/gamelab/animationListModule.js

```diff
@@ -44,12 +44,55 @@ export const START_LOADING_FROM_SOURCE = 'AnimationList/START_LOADING_FROM_SOURC
 export const DONE_LOADING_FROM_SOURCE = 'AnimationList/DONE_LOADING_FROM_SOURCE';
 // Args: {AnimationKey} key, {string} version
 const ON_ANIMATION_SAVED = 'AnimationList/ON_ANIMATION_SAVED';
+// Args: {AnimationKey} key, {!SerializedAnimation} props
+const SET_PENDING_FRAMES = 'AnimationList/SET_PENDING_FRAMES';
+// Args: none
+export const START_LOADING_PENDING_FRAMES_FROM_SOURCE = 'AnimationList/START_LOADING_PENDING_FRAMES_FROM_SOURCE';
+// Args: {AnimationKey} key, {Blob} blob, {String} dataURI. Version?
+export const DONE_LOADING_PENDING_FRAMES_FROM_SOURCE = 'AnimationList/DONE_LOADING_PENDING_FRAMES_FROM_SOURCE';
+// Args: none
+export const REMOVE_PENDING_FRAMES = 'AnimationList/REMOVE_PENDING_FRAMES';
 
 export default combineReducers({
   orderedKeys,
-  propsByKey
+  propsByKey,
+  pendingFrames
 });
 
+/** pendingFrames is used for temporarily storing additional
+  * frames before they get added to the animation in Piskel.
+  * pendingFrames gets added to animation in PiskelEditor.jsx
+  */
+function pendingFrames(state, action) {
+  state = state || {};
+  switch (action.type) {
+
+    case SET_PENDING_FRAMES:
+      return {
+        key: action.key,
+        props: action.props
+      };
+
+    case REMOVE_PENDING_FRAMES:
+      return {};
+
+    case START_LOADING_PENDING_FRAMES_FROM_SOURCE:
+      return Object.assign({}, state, {
+        loadedFromSource: false
+      });
```

Comment:
It's used throughout all of our code. Feel free to not cleanup existing usages if you don't feel like it, but going forward the new syntax is slightly preferred :)